File: NetCore/Lab01/DateUtility.cs
using System;

namespace Lab01;

public class DateUtility
{
    /// <summary>
    /// 已經有人使用，不可以影響到已經用你的人，不可修改參數
    /// 盡可能不要新增 public/internal 的對外介面
    /// </summary>
    /// <returns></returns>
    public bool IsPayday()
    {
        var today = GetToday();

        if (today.Day is 5 or 20)
        {
            return true;
        }

        return false;
    }

    protected virtual DateTime GetToday()
    {
        return DateTime.Today;
    }
}

File: js/src/Lab01/dateUtility.js
export class DateUtility {

    isPayday() {
        const date = this.getToday();

        if (date.getDate() === 5 || date.getDate() === 20) {
            return true;
        }

        return false;
    }

    getToday() {
        const date = new Date();
        return date;
    }
}
